feat: Add an option to override format-time (#2931)

twosmalltrees opened the pull request:
## Description
Addresses #2931, approximately following the suggestions of @gkatsev in the discussion on #3433.  Adds named export `setFormatTime`  to **utils/format-time.js**. Allows overwriting the standard format-time implementation with a custom function, which will be called with seconds and guide as arguments. Also exposes setFormatTime on **videojs**.

Example:

```javascript
videojs.setFormatTime((seconds, guide) => `${seconds}, ${guide}`));
```

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/axedog/edit?html,output))
- [ ] Reviewed by Two Core Contributors

gkatsev commented:
Hey, thanks for your patience. I think this looks good for me. I'll spend a bit more time looking at the code and testing today or tomorrow.

gkatsev commented on src/js/video.js:
```diff
@@ -810,5 +810,10 @@ videojs.dom = Dom;
 videojs.url = Url;
 
+/**
+ * Replaces format-time with a custom implementation
```
can the JSdoc be elaborated here? This is the main doc that Video.js users will see for this method, probably.
Also, we should mention that if an item, like the duration, has already formatted itself, setting a new format time won't update that item.

gkatsev replied:
can this be moved to be below where we set `formatTime` so the two things sit together?

gkatsev commented on src/js/utils/format-time.js:
```diff
@@ -3,20 +3,7 @@
- *         Time formatted as H:MM:SS or M:SS
- */
-function formatTime(seconds, guide = seconds) {
+let implementation = function(seconds, guide) {
```
do we need a way to get back to the original implementation?

pagarwal123 commented:
QA-LGTM

twosmalltrees commented:
No problems at all, thanks for taking the time to review the changes! I've added in `resetFormatTime` to reset back to the default implementation, added to the jsDocs and re-arranged as per your feedback

gkatsev commented:
Thanks @twosmalltrees! It's great to finally get this in!
